### PR TITLE
feat: Add variant list components

### PR DESCRIPTION
**Pull request by willopez**
Add variant list components

### Testing
1. Add multiple variants to a product
2. Verify all variants are rendered
3. Click on a variant and verify URL updates accordingly.

### CODE REVIEW

File: src/routes.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ const routes = require("next-routes")();
 
 routes
   .add("shopProduct", "/shop/:shopSlug/product/:productSlug", "product")
-  .add("product", "/product/:productSlug", "product")
+  .add("product", "/product/:productSlug/:variantId", "product")
```
Comment (mikemurray) [inline src/routes.js]: `:variantId` needs to be optional or there needs to be another route for it as I cannot enter the PDP page without messing with the URL.

File: src/components/ProductDetail/ProductDetail.js [hunk 2]
```diff
@@ -54,6 +55,7 @@ class ProductDetail extends Component {
               pageTitle={catalogProduct.pageTitle}
               title={catalogProduct.title}
             />
+            <VariantList />
```
Comment (mikemurray) [inline src/components/ProductDetail/ProductDetail.js]: Move below `<ProductDetailInfo>`

File: src/components/VariantList/VariantList.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  }
+});
+
+@withStyles(styles)
+class VariantList extends Component {
+  // Default to select first variant
+  state = { activeVariantId: variants[0].node.product._id }
+
+  handleClick = (variant) => {
+    Router.pushRoute("product", { productSlug: variant.slug, variantId: variant._id });
```
Comment (mikemurray) [inline src/components/VariantList/VariantList.js]: Variants don't have slugs. `productSlug` needs to be the `product.slug`

File: src/components/VariantList/VariantList.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    this.setState({ activeVariantId: variant._id });
```
Comment (mikemurray) [inline src/components/VariantList/VariantList.js]: Don't use `setState`, use MobX instead. While it seems like overkill for simple things... it allows us to better track event across the app for analytics.

File: src/components/VariantList/VariantList.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    const { classes: { variantsList } } = this.props;
+    return (
+      <ul className={variantsList}>
+        {variants.map(this.renderVariant)}
+      </ul>
+    );
+  }
+}
+
+VariantList.propTypes = {
```
Comment (mikemurray) [inline src/components/VariantList/VariantList.js]: Put this inside the class as `static propTypes = {}`

File: src/components/VariantItem/VariantItem.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+  render() {
+    const {
+      classes: { variantButton, activeVariant },
+      variant: { title, price },
+      active
+    } = this.props;
+
+    return (
+      <Button
```
Comment (mikemurray) [inline src/components/VariantItem/VariantItem.js]: Consider using `ButtonBase` as it has all the functionality of `Button` but none of the default styles if you're making custom buttons.

File: src/components/VariantItem/VariantItem.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  variantButton: {
+    "display": "flex",
+    "justifyContent": "space-between",
+    "textTransform": "none",
+    "width": "100%",
+    "transition": "opacity 200ms linear",
+    "borderRadius": "2px",
+    "backgroundColor": "#fff",
+    "border": "1px solid #fff",
+    "padding": "10px 20px",
```
Comment (mikemurray) [inline src/components/VariantItem/VariantItem.js]: use some multiple of `theme.spacing.unit` so if we change the global spacing unit, all components will change as well.

File: src/components/VariantItem/VariantItem.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    "&:hover": {
+      border: "1px solid #94e8d1",
+      backgroundColor: "#fff"
```
Comment (mikemurray) [inline src/components/VariantItem/VariantItem.js]: All colors should come from the theme

File: src/components/VariantList/VariantList.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+// Sort by index value
+variants.sort((variant1, variant2) => variant1.index - variant2.index);
+
+const styles = () => ({
+  variantsList: {
+    listStyle: "none",
+    paddingLeft: 0
+  },
+  variantItem: {
+    margin: "10px 0"
```
Comment (mikemurray) [inline src/components/VariantList/VariantList.js]: use `theme.spacing.unit` or some multiple of that.

File: src/components/VariantItem/VariantItem.js [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+import Typography from "material-ui/Typography";
+import classNames from "classnames";
+
+const styles = () => ({
+  variantButton: {
+    "display": "flex",
+    "justifyContent": "space-between",
+    "textTransform": "none",
+    "width": "100%",
+    "transition": "opacity 200ms linear",
```
Comment (mikemurray) [inline src/components/VariantItem/VariantItem.js]: What is this transition for, not seeing it happen?

File: src/components/VariantItem/VariantItem.js [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+        </Typography>
+        <Typography component="span" variant="body1">
+          {price.range}
+        </Typography>
+      </Button>
+    );
+  }
+}
+
+VariantItem.propTypes = {
```
Comment (mikemurray) [inline src/components/VariantItem/VariantItem.js]: `static propTypes = {}` as a member of the class `VariantItem`.

File: src/components/ProductDetail/ProductDetail.js (hunk 2 above)
Comment (willopez) [reply src/components/ProductDetail/ProductDetail.js]: added variants as a prop

**Comment by willopez** [comment]
@mikemurray comments addressed.

**Pull request merged**